### PR TITLE
docs: upgrade --force

### PR DESCRIPTION
**Pull request by serbrech**
**Reason for Change**:

Documents the new `--force` flag for the upgrade command

**Issue Fixed**:

documentation for #522 #525 

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [x] includes documentation

### CODE REVIEW

File: docs/topics/upgrade.md [hunk 1]
```diff
@@ -100,3 +100,24 @@ There are known limitations with VMSS cluster-autoscaler scenarios and upgrade.
...
+## Forcing an upgrade
+
+The upgrade operation takes an optional `--force` argument :
+
+```shellscript
+-f, --force   
+force upgrading the cluster to desired version. Allows same version upgrades and downgrades.
+```
+
+In some situation, you might want to bypass the aks-engine validation on your api-model versions and cluster nodes versions. This is at your own risks and you should assess the potential harm of using this flag.
```
Comment (jackfrancis) [inline docs/topics/upgrade.md]: s/This is at your own risks/This is at your own risk/ (remove trailing "s")

File: docs/topics/upgrade.md [hunk 2]
```diff
@@ -100,3 +100,24 @@ There are known limitations with VMSS cluster-autoscaler scenarios and upgrade.
...
+
+The `--force` parameter instructs the upgrade process to :
+
+- ignore normal version validation
+- include __all__ your cluster's nodes (masters and agents) in the upgrade process. 
+- It will __not__ skip nodes that are already on the target version.
```
Comment (jackfrancis) [inline docs/topics/upgrade.md]: let's indent this point as it's a nice emphasis of the "include all your cluster's nodes" point

File: docs/topics/upgrade.md [hunk 3]
```diff
@@ -100,3 +100,24 @@ There are known limitations with VMSS cluster-autoscaler scenarios and upgrade.
...
+- It will allow any kubernetes versions, including the ones that have not been whitelisted, or deprecated.
```
Comment (jackfrancis) [inline docs/topics/upgrade.md]: Maybe we add an indented line that suggests" `- If you pass in a version that AKS Engine literally cannot install (e.g., a version of Kubernetes that does not exist), you may break your cluster.`

File: docs/topics/upgrade.md [hunk 4]
```diff
@@ -38,7 +38,7 @@ In order to ensure that your `aks-engine upgrade` operation runs smoothly, there
 
 5) `aks-engine upgrade` is backwards compatible. If you deployed with `aks-engine` version `0.27.x`, you can run upgrade with version `0.29.y`. In fact, it is recommended that you use the latest available `aks-engine` version when running an upgrade operation. This will ensure that you get the latest available software and bug fixes in your upgraded cluster.
 
-### Under the hood
+### [Under the hood](#under-the-hood)
```
Comment (mboersma) [inline docs/topics/upgrade.md]: I think this markdown link should be removed. It creates a hyperlink to the same `A HREF` that is on the same line (check the "View File" preview). If you just want to ensure there's an `HREF` for the headline, Markdown does that already.

File: docs/topics/upgrade.md [hunk 5]
```diff
@@ -100,3 +100,24 @@ There are known limitations with VMSS cluster-autoscaler scenarios and upgrade.
 ### Cluster-autoscaler + Availability Set
 
 We don't recommend using `aks-engine upgrade` on clusters that have Availability Set (non-VMSS) agent pools `cluster-autoscaler` at this time.
+
+## Forcing an upgrade
+
+The upgrade operation takes an optional `--force` argument :
```
Comment (mboersma) [inline docs/topics/upgrade.md]: No space goes before a colon (c'est seulement en français).

File: docs/topics/upgrade.md (hunk 1 above)
Comment (mboersma) [inline docs/topics/upgrade.md]: In some situation**s** (phrase should end with an "s')

File: docs/topics/upgrade.md [hunk 6]
```diff
@@ -100,3 +100,24 @@ There are known limitations with VMSS cluster-autoscaler scenarios and upgrade.
...
+
+```shellscript
+-f, --force   
+force upgrading the cluster to desired version. Allows same version upgrades and downgrades.
+```
+
+In some situation, you might want to bypass the aks-engine validation on your api-model versions and cluster nodes versions. This is at your own risks and you should assess the potential harm of using this flag.
+
+The `--force` parameter instructs the upgrade process to :
```
Comment (mboersma) [inline docs/topics/upgrade.md]: No space before colon again.

File: docs/topics/upgrade.md (hunk 2 above)
Comment (mboersma) [inline docs/topics/upgrade.md]: Items in a bulleted list should follow parallel construction. So if one is a complete sentence, they should all be complete sentences, or if one starts with a verb, they all should, and so on. So maybe change the last three to something like:
```
- __not__ to skip nodes that are already on the target version
- allow any Kubernetes versions, ....
- accept downgrade operations
```

(Sorry to be a pedant. I just read [_Dreyer's English_](https://www.goodreads.com/book/show/40063024-dreyer-s-english) and I'm all fired up.)

File: docs/topics/upgrade.md (hunk 3 above)
Comment (mboersma) [inline docs/topics/upgrade.md]: Kubernetes should be capitalized.

File: docs/topics/upgrade.md [hunk 7]
```diff
@@ -100,3 +100,24 @@ There are known limitations with VMSS cluster-autoscaler scenarios and upgrade.
...
+
+- ignore normal version validation
+- include __all__ your cluster's nodes (masters and agents) in the upgrade process. 
+- It will __not__ skip nodes that are already on the target version.
+- It will allow any kubernetes versions, including the ones that have not been whitelisted, or deprecated.
+- It will accept downgrade operations.
+
+For each nodes, the cluster will follow the same process as described in the section above: [Under the hood](#under-the-hood)
```
Comment (mboersma) [inline docs/topics/upgrade.md]: Oh, now I get it. I think the reference `#under-the-hood` will work without you changing anything about the header above.

File: docs/topics/upgrade.md (hunk 2 above)
Comment (tariq1890) [reply docs/topics/upgrade.md]: product placement!

File: docs/topics/upgrade.md (hunk 1 above)
Comment (tariq1890) [inline docs/topics/upgrade.md]: It should be `in your api-model `

File: docs/topics/upgrade.md [hunk 8]
```diff
@@ -100,3 +100,24 @@ There are known limitations with VMSS cluster-autoscaler scenarios and upgrade.
...
+```shellscript
+-f, --force   
+force upgrading the cluster to desired version. Allows same version upgrades and downgrades.
+```
+
+In some situation, you might want to bypass the aks-engine validation on your api-model versions and cluster nodes versions. This is at your own risks and you should assess the potential harm of using this flag.
+
+The `--force` parameter instructs the upgrade process to :
+
+- ignore normal version validation
```
Comment (tariq1890) [inline docs/topics/upgrade.md]:
```suggestion
- bypass the usual version validation.
```

Comment (tariq1890) [reply docs/topics/upgrade.md]: Actually, even normal works. But I still recommend the rest of the changes in my GH suggestion.

File: docs/topics/upgrade.md (hunk 2 above)
Comment (serbrech) [reply docs/topics/upgrade.md]: Ha, I actually thought exactly that, but felt it was better expressed with a full sentence.
Fixing it anyway. My sense for proper english isn't good enough to dispute Dreyer's English :)

File: docs/topics/upgrade.md (hunk 1 above)
Comment (serbrech) [reply docs/topics/upgrade.md]: or rather `of`? :
>you might want to bypass the aks-engine validation _of_ your api-model versions and cluster nodes versions

File: docs/topics/upgrade.md [hunk 9]
```diff
@@ -100,3 +100,26 @@ There are known limitations with VMSS cluster-autoscaler scenarios and upgrade.
...
+
+The `--force` parameter instructs the upgrade process to:
+
+- bypass the usual version validation
+- include __all__ your cluster's nodes (masters and agents) in the upgrade process
+  -  nodes that are already on the target version will __not__ be skipped.
+- allow any Kubernetes versions, including the ones that have not been whitelisted, or deprecated
+- accept downgrade operations
+
+> Note : If you pass in a version that AKS-Engine literally cannot install (e.g., a version of Kubernetes that does not exist), you may break your cluster.
```
Comment (mboersma) [inline docs/topics/upgrade.md]: No space before a colon.

File: docs/topics/upgrade.md [hunk 10]
```diff
@@ -100,3 +100,26 @@ There are known limitations with VMSS cluster-autoscaler scenarios and upgrade.
...
+
+For each nodes, the cluster will follow the same process as described in the section above: [Under the hood](#under-the-hood)
```
Comment (mboersma) [inline docs/topics/upgrade.md]: "For each node," (node should be singular). Also the word `as` can be deleted: "follow the same process described in..."

**Pull request merged**